Save VASP stresses as matrix

pmrv opened the pull request:
Turns out VASP writes the stress components in non-Voigt order (*WHY?!*) and I missed that in #354 originally.  This provides the fix.

@niklassiemer This is probably worth pointing out in the next CHANGELOG.

pmrv committed:
Save VASP stresses as matrix

liamhuber commented:
Am I Right to understand this will resolve those weird artifacts in the ml project strain plots?

pmrv commented:
> Am I Right to understand this will resolve those weird artifacts in the ml project strain plots?

I very much hope so, but I'll need to regenerate some data to say for sure.

pmrv committed:
WIP: restructure test

pmrv committed:
Clean up tests

Expands the tests to check against all OUTCARs that we have, also ones
with multiples stresses, off-diagonals and additional vdW contributions
that break the old parsing code

pmrv committed:
Remove fixed offsets when parsing stresses

If there are additional stress contributions, like vdW this would break

pmrv commented:
I added up needed to rewire things a bit more.  I've cleaned up the tests to check against all our OUTCARs which exposed a latent bug in the old parsing that I've patched over.

pmrv commented:
Also as discussed in the pyiron meeting, instead of saving as Voigt notation I save the full matrix now that same we do for lammps.

sudarsan-surendralal commented on tests/vasp/test_outcar.py:
```diff
@@ -32,6 +32,10 @@ def setUpClass(cls):
             filename = posixpath.join(direc, f)
             cls.file_list.append(filename)
 
+    def setUp(self):
```
This is pretty neat! Maybe we could move this (in a separate PR) to `pyiron_base._tests` so that all pyiron unittests inherit this

niklassiemer replied:
This enables `self.assertEqual(np_array1, np.array2)`? Then I really like your suggestion to make it available more broadly!

sudarsan-surendralal reviewed (approved):
LGTM!
